server/src/main/java/org/apache/iotdb/db/auth/StandaloneAuthorityFetcher.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.auth;
+
+import org.apache.iotdb.common.rpc.thrift.TSStatus;
+import org.apache.iotdb.commons.auth.AuthException;
+import org.apache.iotdb.commons.utils.AuthUtils;
+import org.apache.iotdb.confignode.rpc.thrift.TAuthorizerReq;
+import org.apache.iotdb.db.client.ConfigNodeClient;
+import org.apache.iotdb.db.localconfignode.LocalConfigNode;
+import org.apache.iotdb.db.mpp.plan.execution.config.ConfigTaskResult;
+import org.apache.iotdb.rpc.RpcUtils;
+import org.apache.iotdb.rpc.TSStatusCode;
+
+import com.google.common.util.concurrent.SettableFuture;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class StandaloneAuthorityFetcher implements IAuthorityFetcher {
+
+  private static final Logger logger = LoggerFactory.getLogger(StandaloneAuthorityFetcher.class);
+
+  private LocalConfigNode localConfigNode = LocalConfigNode.getInstance();
+
+  private static final class StandaloneAuthorityFetcherHolder {
+    private static final StandaloneAuthorityFetcher INSTANCE = new StandaloneAuthorityFetcher();
+
+    private StandaloneAuthorityFetcherHolder() {}
+  }
+
+  public static StandaloneAuthorityFetcher getInstance() {
+    return StandaloneAuthorityFetcher.StandaloneAuthorityFetcherHolder.INSTANCE;
+  }
+
+  @Override
+  public TSStatus checkUser(String username, String password) {
+    try {
+      if (localConfigNode.login(username, password)) {
+        return RpcUtils.getStatus(TSStatusCode.SUCCESS_STATUS);
+      } else {
+        return RpcUtils.getStatus(
+            TSStatusCode.WRONG_LOGIN_PASSWORD_ERROR, "Authentication failed.");
+      }
+    } catch (AuthException e) {
+      return RpcUtils.getStatus(TSStatusCode.AUTHENTICATION_ERROR, e.getMessage());
+    }
+  }
+
+  @Override
+  public TSStatus checkUserPrivileges(String username, List<String> allPath, int permission) {
+    boolean checkStatus = true;
+    String checkMessage = null;
+    for (String path : allPath) {
+      try {
+        if (!checkOnePath(username, path, permission)) {
+          checkStatus = false;
+        }
```

Comment:
if return false when checking one path, we should just break the loop.

for example, if we have 3 paths,the user have no permission  in the second path, in this current implemention, the final result of `checkStatus` is true.

Reply:
fixed~